src/Frontend/Core.hs

```diff
@@ -282,35 +310,36 @@ instance (KnownSymbol s, Page a) => Page (Capture s c :> a) where
     isAuthorized = error "IMPOSSIBLE: instance (KnownSymbol s, Page a) => Page (Capture s c :> a)"
 
 instance Page a => Page (Headers h a) where
-    isAuthorized = error "IMPOSSIBLE: instance Page a => Page (Headers h a)"
+    isAuthorized = isAuthorized . fmap getResponse
+    extraPageHeaders = extraPageHeaders . getResponse
 
 instance (KnownSymbol s, Page a) => Page (QueryParam s b :> a) where
     isAuthorized = error "IMPOSSIBLE: instance (KnownSymbol s, Page a) => Page (QueryParam s b :> a)"
 
 instance Page a => Page (FormReqBody :> a) where
     isAuthorized = error "IMPOSSIBLE: instance Page a => Page (FormReqBody :> a)"
 
-instance Page a => Page (Get c a) where
-    isAuthorized = error "IMPOSSIBLE: instance Page a => Page (Get c a)"
+-- We must not generalize this instance to a `Page (Get c a)` instance.
+instance Page a => Page (Get c (GetResult a)) where
+    isAuthorized = error "IMPOSSIBLE: instance Page a => Page (Get c (GetResult a))"
+
+-- We must not generalize this instance to a `Page (Get c a)` instance.
+instance Page a => Page (Get c (Frame a)) where
```

Comment:
does this mean that `Frame` is part of the security mechanism?  i think i'm asking weird questions.  bear with me.  (-: